src/WebJobs.Script.WebHost/DependencyInjection/JobHostScopedServiceProviderFactory.cs

```diff
@@ -76,12 +76,69 @@ public IServiceProvider CreateServiceProvider(IServiceCollection services)
             return jobHostServices.BuildServiceProvider();
         }
 
+        private void ShimBreakingChanges(IServiceCollection services, ILogger logger)
+        {
+            ShimActivatorUtilitiesConstructorAttributeTypes(services, logger);
+            ShimLegacyNetWorkerMetadataProvider(services, logger);
+        }
+
+        /// <summary>
+        /// Versions 1.3.0 and older of Microsoft.Azure.Functions.Worker.Sdk register a type with a constructor that
+        /// takes a strign parameter. This workeed with DryIoc, but not with the new DI container.
+        /// Shimming those types to provide backwards compatibility, but this should be removed in the future.
+        /// </summary>
+        /// <param name="services">The <see cref="ServiceCollection"/> to inspect and modify.</param>
+        private static void ShimLegacyNetWorkerMetadataProvider(IServiceCollection services, ILogger logger)
```

Comment:
nit:  missing `logger` in method summary.

Reply:
Good catch. This was a late addition. Also updated `ShimActivatorUtilitiesConstructorAttributeTypes` as I added the parameter there as well.